executor/adapter.go

```diff
@@ -1485,10 +1485,31 @@ func (a *ExecStmt) LogSlowQuery(txnTS uint64, succ bool, hasMoreResults bool) {
 	cfg := config.GetGlobalConfig()
 	costTime := time.Since(sessVars.StartTime) + sessVars.DurationParse
 	threshold := time.Duration(atomic.LoadUint64(&cfg.Instance.SlowThreshold)) * time.Millisecond
+	expensiveRatio := atomic.LoadUint64(&cfg.Instance.ExpensiveScanRatio)
 	enable := cfg.Instance.EnableSlowLog.Load()
+
+	// Check if it's an expensive mvcc scan. It's regarded as expensive if
+	// 1. The rocksdb_delete_skipped_count > processed_keys * expensive_ratio
+	// 2. The processed_keys is 0 but the duration is larger than slow threshold.
+	execDetail := stmtCtx.GetExecDetails()
+	isExpensiveScan := false
+	if execDetail.ScanDetail != nil && sessVars.ConnectionID > 0 {
+		processedKeys := execDetail.ScanDetail.ProcessedKeys
+		if processedKeys == 0 {
```

Comment:
I guess maybe it can treat processedKeys == 0 as if processedKeys == 1 and the the same logic below. 
Otherwise it's unreasonable that one query with 1 processedKeys and 200 scan is marked as expensive scan but another query with 0 processedKeys and 400 scan is not.

Reply:
I thinj about the solution first, there could be another situation where `processedKeys == 0` and the number of iterated keys are 20, these queries could be recorded.
Both of them have some drawbacks.

Reply:
Changed to the unifying method when `processedKeys == 0`, PTAL